### PR TITLE
VM integration tests

### PR DESCRIPTION
**Pull request by kunagpal**

### CODE REVIEW

File: npm/test-vm.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+        // specially load bootstrap file
+        mocha.addFile(path.join(SPEC_SOURCE_DIR, '_bootstrap.js'));
+
+        files.filter(function (file) { // extract all test files
+            return (file.substr(-8) === '.test.js');
+        }).forEach(mocha.addFile.bind(mocha));
+
+        // run loaded tests in the Node VM
+        vm.runInContext(`mocha.run(function (err) {
```
Comment (czardoz) [inline npm/test-vm.js]: This `require`s mocha from the outside the VM. All the sandbox require's are processed outside the VM, so this doesn't truly test the functionality inside the VM. 

Ideally, what you want is:

1. Create a VM context
2. Run the bundle in the context
3. Then load and run tests in the same VM context

Comment (kunagpal) [inline npm/test-vm.js]: Done.

File: npm/test-vm.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            }).forEach(mocha.addFile.bind(mocha));
+    
+            mocha.run(function (err) {
+                err && console.error(err.stack || err);
+                exit(err ? 1 : 0);
+            });
+        });
+    `, new vm.createContext({
+        exit: exit,
+        require: require,
```
Comment (shamasis) [inline npm/test-vm.js]: This is not correct. If you require with a require function from outside, it is actually required outside VM. If this worked, we would have never needed browserify in sandbox.

Comment (kunagpal) [inline npm/test-vm.js]: Fixed

**Pull request merged**